reana-ui/src/pages/launchOnReana/LaunchOnReana.js

```diff
@@ -75,37 +81,102 @@ export default function LaunchOnReana() {
+     */
+    function fn(query) {
+      try {
+        return JSON.parse(query.get("parameters"));
+      } catch (error) {
+        dispatch(
+          triggerNotification(
+            "An error has occurred",
+            `Invalid JSON workflow parameters provided: ${query.get(
+              "parameters"
+            )}`,
+            { error: true }
+          )
+        );
+      }
+      return null;
+    }
+    return fn(query);
+  }, [dispatch, query]);
+
   return (
     <BasePage>
       <Container text className={styles["container"]}>
         <div>
           <Title>Launch on REANA</Title>
         </div>
         {isMissingRequiredParams(query) ? (
-          <Message
-            icon="warning circle"
-            header="Missing required query string fields."
-            content={`The required fields: ${getRequiredQsParams()}`}
-            warning
-          />
+          <Welcome />
         ) : (
-          <ul>
-            {[...query].map(([k, v]) => (
-              <li key={k}>
-                {k}: {v}
-              </li>
-            ))}
-          </ul>
+          <Box className={styles.box} wrap>
+            <section className={styles.details}>
+              <span className={styles.icon}>
+                <Icon name="hourglass half" />
+              </span>
+              <div>
+                <div className={styles.name}>
+                  {query.get("name") ?? DEFAULT_WORKFLOW_NAME}
+                </div>
+                <a className={styles.url} href={query.get("url")}>
+                  {query.get("url")}
+                </a>
+                {query.get("parameters") && getWorkflowParameters() && (
+                  <div className={styles.parameters}>
+                    Parameters:
+                    <ul>
+                      {Object.entries(getWorkflowParameters()).map(([k, v]) => (
+                        <li key={k}>
+                          {k}: {v}
+                        </li>
+                      ))}
+                    </ul>
+                  </div>
+                )}
+              </div>
+            </section>
+            <section className={styles.actions}>
+              <Button
+                primary
+                icon="rocket"
+                content="Launch"
+                onClick={handleLaunch}
+                disabled={loading || isMissingRequiredParams(query)}
+              />
+            </section>
+            <BadgeEmbed />
+          </Box>
         )}
-        <Button
-          primary
-          icon="rocket"
-          content="Launch"
-          onClick={handleLaunch}
-          disabled={loading || isMissingRequiredParams(query)}
-        />
         {loading && <Loader content="Executing workflow..." />}
       </Container>
     </BasePage>
   );
 }
+
+const BadgeEmbed = () => (
+  <details className={styles.badge}>
+    <summary>
+      Expand to see the text below, paste it into your README to show a REANA
```

Comment:
I worry that it may be confusing some users to see two "Launch" actionable buttons, as it were, close to each other:

![l1](https://user-images.githubusercontent.com/517546/158435245-0cfe76e5-41e2-4984-a9bd-ef63fe04ef29.png)

It could provide some visual clutter as to "where to click"?

Especially because there will be few badge creators (=maintainers of analysis repositories), but possibly very many badge users (=students clicking on existing badges). For the latter group, they only want to see the analysis run, they are not interested in creating a new badge -- especially if they came to the launcher web site via an already-existing badge produced by the creator.

Several possible solutions how to alleviate this impression:

1. Simplify the badge phrase into something very short like "See Markdown source code for a launch badge:" AND not show how the badge would look like so close up-there AND move the badge source part outside the workflow box at the top to somewhere near the bottom of the page to make it visually very separate.
2. Detect referrer and show the badge source functionality only if we can detect that people are just editing URL on the fly vs people coming to this launcher page via some nice referrer page such as GitHub or Zenodo.
3. Hide the functionality completely for now, whilst we are in the R/O mode, and plug it back once we shall be in the R/W mode offering editable URL boxes either here on this page or on a dedicated separate "badge generator" page.

Reply:
I went for option 3 for now.